interface/src/App.tsx

```diff
@@ -1,75 +1,73 @@
-import React, { Component } from 'react'
+import React, { useEffect, useState } from "react"
 import { BrowserRouter, Route } from "react-router-dom"
-import _ from 'lodash'
+import _ from "lodash"
 import HeaderBar from "./components/lib/header-bar.js"
 
-import { ThemeProvider } from 'styled-components'
+import { ThemeProvider } from "styled-components"
 
-import light from './components/themes/light'
-import dark from './components/themes/dark'
+import { CalendarAndEventProvider } from "./hooks/useCalendarsAndEvents"
+import { UserLocationProvider } from "./hooks/useUserLocation"
 
-import { Box } from '@tlon/indigo-react'
-import CalendarWrapper from './views/CalendarWrapper'
-import Store, { StoreState } from './logic/store'
-import CalendarView from './views/CalendarView'
-import EventView from './views/EventView'
-import Actions from './logic/actions'
-import UrbitApi from './logic/api'
-import Subscription from './logic/subscription'
+import lightTheme from "./components/themes/light"
+import darkTheme from "./components/themes/dark"
 
-interface Props {}
+import { Box } from "@tlon/indigo-react"
+import CalendarWrapper from "./views/CalendarWrapper"
+import CalendarView from "./views/CalendarView"
+import EventView from "./views/EventView"
 
-interface State extends StoreState {
-  dark: boolean
-}
-
-export class App extends Component<Props, State> {
-  themeWatcher: any
-  store: Store
-  actions: Actions
-  api: UrbitApi
-  subscription: Subscription
-
-  constructor(props) {
-    super(props)
-
-    this.store = new Store()
-    this.store.setStateHandler(this.setState.bind(this))
-    this.state = { ...this.store.state, dark: false }
-    this.api = new UrbitApi()
-    this.actions = new Actions(this.store, this.api)
-
-    this.subscription = new Subscription(this.store, this.actions, this.api)
-  }
-
-  updateTheme = (updateTheme) : void => {
-    this.setState({ dark: updateTheme })
-  }
+export const App: React.FC = () => {
+  const [dark, setDark] = useState(false)
+  let themeWatcher: any
 
-  componentDidMount() {
-    this.themeWatcher = window.matchMedia('(prefers-color-scheme: dark)')
-    this.setState({ dark: this.themeWatcher.matches })
-    this.themeWatcher.addListener(this.updateTheme)
+  const updateTheme = (updateTheme): void => {
+    setDark(updateTheme)
   }
 
-  render() {
-    const { state } = this
+  useEffect(() => {
+    themeWatcher = window.matchMedia("(prefers-color-scheme: dark)")
+    setDark(themeWatcher.matches)
+    themeWatcher.addListener(updateTheme)
+  })
```

Comment:
Is this only supposed to run on the initial mount? And if so, should it have an empty dependency array after the anonymous function?

Reply:
Agreed, this should have an empty dependency array.